### PR TITLE
feat: Constraint minimal replicas

### PR DESCRIPTION
**Pull request by a-thaler**
## Description

Changes proposed in this pull request:

- Improved API description for manual scaling
- Added minimum constraint

## Related Issues and Documents

Closes:

Related issues:
https://github.com/kyma-project/telemetry-manager/issues/430

## Traceability

- [x] The PR is linked to a GitHub Issue.
- [x] New features have a milestone label set.
- [x] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [x] The corresponding GitHub Issue has a respective `area` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.

## Testability

The feature is unit-tested:

- [x] Yes.
- [ ] No, because unit tests are not needed.
- [ ] No, because of ...

The feature is e2e-tested:

- [x] Yes.
- [ ] No, because e2e-tests are not needed.
- [ ] No, because of ...

Tests conducted for the PR:

## Codebase

- [ ] My code follows the [Effective Go](https://go.dev/doc/effective_go) style guidelines.
- [ ] The code was planned and designed following the defined architecture and the separation of concerns.
- [ ] The code has sufficient comments, particularly for all hard-to-understand areas.
- [ ] This PR adds value and shows no feature creep.
- [ ] I have augmented the test suite that proves my fix is effective or that my feature works.
- [x] Adjusted the documentation if the change is user-facing.

### CODE REVIEW

File: apis/operator/v1alpha1/telemetry_types.go [hunk 1]
```diff
@@ -63,15 +63,15 @@ type TraceGatewaySpec struct {
 	Scaling Scaling `json:"scaling,omitempty"`
 }
 
-// Scaling defines the scaling type and number of pods.
+// Scaling defines which strategy will be used for scaling the gateway with detailed configuration options per strategy type.
```
Comment (NHingerl) [inline apis/operator/v1alpha1/telemetry_types.go:66]:
```suggestion
// Scaling defines which strategy is used for scaling the gateway, with detailed configuration options for each strategy type.
```

File: apis/operator/v1alpha1/telemetry_types.go [hunk 2]
```diff
@@ -85,7 +85,8 @@ const (
 )
 
 type StaticScaling struct {
-	// Replicas defines the number of pod instances.
+	// Replicas defines a static number of pods to use for running the gateway. Minimum is 1.
```
Comment (NHingerl) [inline apis/operator/v1alpha1/telemetry_types.go:88]:
```suggestion
	// Replicas defines a static number of pods to run the gateway. Minimum is 1.
```

File: test/e2e/metrics_basic_test.go [hunk 3]
```diff
@@ -27,7 +27,7 @@ import (
 	"github.com/kyma-project/telemetry-manager/test/testkit/verifiers"
 )
 
-var _ = Describe("Metrics Basic", Label("metrics"), func() {
+var _ = Describe("Metrics Basic", Label("icke"), func() {
```
Comment (NHingerl) [inline test/e2e/metrics_basic_test.go:30]: :D

**Review by NHingerl — DISMISSED** [review]

File: test/e2e/metrics_basic_test.go (hunk 3 above)
Comment (a-thaler) [reply test/e2e/metrics_basic_test.go:30]: ahh, good finding :)

**Review by NHingerl — APPROVED** [review]

**Review by NHingerl — APPROVED** [review]

**Review by skhalash — APPROVED** [review]

**Pull request merged**